src/dataworker/index.ts

```diff
@@ -162,7 +162,11 @@ export async function runDataworker(_logger: winston.Logger, baseSigner: Wallet)
         logger[startupLogLevel(config)]({ at: "Dataworker#index", message: "Executor disabled" });
       }
 
-      await clients.multiCallerClient.executeTransactionQueue();
+      // Execute a series of transactions to reflect the state changes from the above operations.
+      // This is done only if the sendingDataworkerTxsEnabled flag is set to true.
+      if (config.transactionExecutionsEnabled) {
+        await clients.multiCallerClient.executeTransactionQueue();
+      }
```

Comment:
If you pass in the argument then it'll toggle simulation mode in the `MultiCallerClient`. This is what is done on the relayer side.

```suggestion
        await clients.multiCallerClient.executeTransactionQueue(!config.transactionExecutionsEnabled);
```